src/components/feedback/Dialog.tsx

```diff
@@ -48,19 +49,33 @@ type ComponentProps = {
    * Optional TransitionComponent for open (mount) and close (unmount) transitions
    */
   transitionComponent?: TransitionComponent;
-
-  /**
-   * Dialog lays out its content in a Panel by default, but `custom` allows any
-   * layout of Dialog children
-   */
-  variant?: 'panel' | 'custom';
 };
 
 // This component forwards a number of props on to `Panel` but always sets the
 // `fullWidthHeader` prop to `true`.
-export type DialogProps = PresentationalProps &
+export type PanelDialogProps = PresentationalProps &
   ComponentProps &
-  Omit<PanelProps, 'fullWidthHeader'>;
+  Omit<PanelProps, 'fullWidthHeader'> & {
+    /** Dialog lays out its content in a Panel by default */
+    variant?: 'panel';
+  };
+
+type NonPanelDialogProps = PresentationalProps &
```

Comment:
Maybe name this `CustomDialogProps` to match the `variant` value?

Reply:
Makes sense